rflx/generator/allocator.py

```diff
@@ -236,31 +260,24 @@ def _create_init_proc(self) -> UnitPart:
     def _needs_allocation(type_: rty.Type) -> bool:
         return isinstance(type_, (rty.Message, rty.Sequence))
 
-    def _create_slot_with_size(self, size: int) -> int:
-        slot_num = len(self._slot_sizes) + 1
-        self._slot_sizes[slot_num] = size
-        return slot_num
-
-    def _map_location_to_slot(self, location: Optional[Location], slot: int) -> None:
-        assert location is not None
-        self._allocation_slots[location] = slot
-
-    def _allocate_global_slots(self) -> None:
+    def _allocate_global_slots(self) -> List[SlotInfo]:
+        slots = []
         for d in self._session.declarations.values():
             if isinstance(d, decl.VariableDeclaration) and self._needs_allocation(d.type_):
-                slot = self._create_slot_with_size(self.get_size(d.identifier))
-                self._map_location_to_slot(d.location, slot)
-                self._global_slots.add(slot)
+                assert d.location is not None
+                slots.append(SlotInfo(self.get_size(d.identifier), [d.location]))
+        return slots
 
-    def _allocate_local_slots(self) -> None:
+    def _allocate_local_slots(self) -> List[SlotInfo]:
         """
         Allocate slots for state variables and state actions.
 
         We collect the allocation requirements (pairs of locations and sizes) for each
-        state in a list, then sort these lists in descending order of sizes, filling up
-        with zero sizes to make the lists the same length. For each list index, we compute
-        the maximal requested size for that index. Finally, we create a memory slot for
-        each index and map the locations to those slots.
+        state in a list, then sort these lists in descending order of sizes.
+        Then, we imagine a matrix where each line is one of these lists (filling
+        up with zero sizes to make the lists the same length)). For each column of
```

Comment:
Double closing parenthesis.

Reply:
fixed in dc8838412927382bf2e36f4042733b98bab41d90